graphql_api/views.py

```diff
@@ -290,6 +305,40 @@ def _get_user(self, request):
         if request.user:
             request.user.pk
 
+    def _check_ratelimit(self, request):
+        redis = get_redis_connection()
+        user_ip = self.get_client_ip(request)
+        try:
+            # eagerly try to get user_id from request object
+            user_id = request.user.pk
+        except AttributeError:
+            user_id = None
+
+        if user_id:
+            key = f"rl-user:{user_id}"
+        else:
+            key = f"rl-ip:{user_ip}"
```

Comment:
Can we just fetch the ip here? Seems like we don't use it unless the user id does not exist

Reply:
Good point, will make the change